### PR TITLE
add slather in fast file"

### PR DESCRIPTION
**Pull request by huminzhi**

### CODE REVIEW

File: libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m [hunk 1]
```diff
@@ -418,6 +418,9 @@ - (SFSyncState*)syncUpWithTarget:(SFSyncUpTarget *)target
                          options:(SFSyncOptions *)options
                         soupName:(NSString *)soupName
                      updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (target == nil) {
```
Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m]: Who is calling this with nil? How was it working before?
I think something is failing before the call to this method (maybe a regression introduced by my recent changes https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1892). I think this change is just masking the issue.

Comment (bhariharan) [inline libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m]: +1 to what Wolf said. I don't think `target` should be `nil` here.

Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m]: My earlier PR (https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1892) is indeed the cause of the regression. I'll have a PR to fix the issue shortly.

Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m]: This should fix hybrid smartsync tests: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1901

**Comment by huminzhi** [comment]
@wmathurin reverted my changes in this pr

**Comment by huminzhi** [comment]
@helenren  I added slather in the fastlane file, we should get more details code cov info now..

**Pull request merged**